File: rapi/src/main/kotlin/me/gachalyfe/rapi/domain/model/AnomalyInterception.kt
package me.gachalyfe.rapi.domain.model

import java.util.Optional

data class AnomalyInterception(
    val id: Long,
    val date: String,
    val bossName: String,
    val stage: Int,
    val dropType: String,
    val dropped: Boolean,
    val modules: Int,
    val equipment: Optional<ManufacturerEquipment> = Optional.empty(),
)

File: gl-frontend/src/models/domain/AnomalyInterception.ts
import { ManufacturerEquipment } from './ManufacturerEquipment'

export interface AnomalyInterception {
  id?: number
  date: string
  bossName: string
  stage: number
  dropType: string
  dropped: boolean
  modules: number
  equipment?: ManufacturerEquipment
}
